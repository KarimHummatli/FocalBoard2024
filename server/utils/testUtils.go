package utils

import "github.com/stretchr/testify/mock"

var Anything = mock.MatchedBy(func(any) bool { return true })
