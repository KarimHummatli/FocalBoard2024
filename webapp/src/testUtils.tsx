// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {IntlProvider} from 'react-intl'
import React from 'react'

export const wrapIntl = (children: any) => <IntlProvider locale='en'>{children}</IntlProvider>
