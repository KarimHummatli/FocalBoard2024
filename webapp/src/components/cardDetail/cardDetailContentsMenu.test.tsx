// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import {Provider as ReduxProvider} from 'react-redux'

import {TestBlockFactory} from '../../test/testBlockFactory'

import {wrapIntl, mockStateStore} from '../../testUtils'

import CardDetailContentsMenu from './cardDetailContentsMenu'

const board = TestBlockFactory.createBoard()
const card = TestBlockFactory.createCard(board)
describe('components/cardDetail/cardDetailContentsMenu', () => {
    // const state = {
    //     users: {
    //         me: {
    //             id: 'user-id-1',
    //             username: 'username_1',
    //         },
    //     },
    // }
    const store = mockStateStore([], {})
    beforeEach(() => {
        jest.clearAllMocks()
    })
    test('return cardDetailContentsMenu', () => {
        const {container} = render(wrapIntl(
            <ReduxProvider store={store}>
                <CardDetailContentsMenu card={card}/>
            </ReduxProvider>,
        ))
        const buttonElement = screen.getByRole('button', {name: 'menuwrapper'})
        userEvent.click(buttonElement)
        expect(container).toMatchSnapshot()
    })

    test('return cardDetailContentsMenu and add content', () => {
        const {container} = render(wrapIntl(
            <ReduxProvider store={store}>
                <CardDetailContentsMenu card={card}/>
            </ReduxProvider>,
        ))
        screen.debug()
        const buttonElement = screen.getByRole('button', {name: 'menuwrapper'})
        userEvent.click(buttonElement)
        screen.debug()
        const buttonAdd = screen.getByRole('button', {name: 'Add content'})
        userEvent.click(buttonAdd)
        screen.debug()
        expect(container).toMatchSnapshot()
    })
})
