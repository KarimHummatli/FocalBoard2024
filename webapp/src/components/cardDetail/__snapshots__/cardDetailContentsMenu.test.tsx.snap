// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/cardDetail/cardDetailContentsMenu return cardDetailContentsMenu 1`] = `
<div>
  <div
    class="CardDetailContentsMenu content add-content"
  >
    <div
      aria-label="menuwrapper"
      class="MenuWrapper"
      role="button"
    >
      <button
        class="Button    "
        type="button"
      >
        <span>
          Add content
        </span>
      </button>
      <div
        class="Menu noselect top"
      >
        <div
          class="menu-contents"
        >
          <div
            class="menu-options"
          />
          <div
            class="menu-spacer hideOnWidescreen"
          />
          <div
            class="menu-options hideOnWidescreen"
          >
            <div
              aria-label="Cancel"
              class="MenuOption TextOption menu-option menu-cancel"
              role="button"
            >
              <div
                class="noicon"
              />
              <div
                class="menu-name"
              >
                Cancel
              </div>
              <div
                class="noicon"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/cardDetail/cardDetailContentsMenu return cardDetailContentsMenu and add content 1`] = `
<div>
  <div
    class="CardDetailContentsMenu content add-content"
  >
    <div
      aria-label="menuwrapper"
      class="MenuWrapper"
      role="button"
    >
      <button
        class="Button    "
        type="button"
      >
        <span>
          Add content
        </span>
      </button>
    </div>
  </div>
</div>
`;
